inst/private/check_and_convert.m

```diff
@@ -29,10 +29,22 @@
         'if isinstance(temp, sp.Matrix) and temp.shape == (1, 1):',
        ['    ' cur_var ' = temp[0, 0]']}, newl));
 
-
+    is_dict_with_sym_keys_curvar = pyeval(['isinstance(', cur_var, ', dict) and len(' cur_var '.keys())>0 and isinstance(' cur_var, '.keys()[0], (sp.Basic, sp.MatrixBase))']);
     is_list_curvar = pyeval(['isinstance(', cur_var, ', (list, tuple))']);
+
     if is_list_curvar
       obj{i} = check_and_convert(cur_var);
+    elseif is_dict_with_sym_keys_curvar
+      %if cur_var is dictionary with symbols as keys then convert it to a struct
+      pyexec(strjoin({'_allKeysStr = []',
+                      '_allValues = []',
+                      ['for key, value in ' cur_var '.iteritems():'],
+                      '    _allKeysStr.append(str(key))',
+                      '    _allValues.append(value)'}, newl));
+      _allKeysStr = pyeval('_allKeysStr');
```

Comment:
Not allowed in Matlab.  I think github does not know about Octave.  That's sometimes a problem for octsympy, if Matlab even parses one of our files but I don't think it will get to this one, so don't worry about it.